dts/arm/st/stm32f0.dtsi

```diff
@@ -55,11 +62,92 @@
 			label = "STM32_CLK_RCC";
 		};
 
+#ifdef CONFIG_PINCTRL_STM32
```

Comment:
@erwango @galak @superna9999 
> I don't think we'll keep and maintain 2 variants of dts

With the #ifdef's one can still choose between the pinmux and the pinctrl driver. If I remove them the pinctrl driver becomes mandatory for STM32F0 SoCs. Would that be acceptable?

Reply:
Well, maybe other STM32 users have their voice on this (@superna9999, @ydamigos, @DWagenk , @vaussard, @lowlander, ..), but in the end I think we should focus on one driver.
As already mentioned for SPI in previous comment, maintaining one driver and one version of DTS is a lot of work. If we think pinctrl driver is a good alternative to existing implementation, let's switch to it. 
What could be acceptable is to maintain 2 versions the time needed to switch whole STM32 series to this new driver, but it could only be temporary.
Zephyr is still quite young and moving so it is still time to make this kind of radical changes if we think this is the good direction, I think that people would understand it. Though, having 2 possible drivers dedicated to the same component would be confusing for newcomers.